app/javascript/components/visualization/DotPlot.jsx

```diff
@@ -18,6 +20,166 @@ export const dotPlotColorScheme = {
   values: [0, 0.5, 1]
 }
 
+/**
+ * Adds rudimentary service worker cache optimization to Morpheus
+ *
+ * Without SWC, dot plots can take prohibitively long to load in local development
+ * for realistic datasets.
+ */
+function patchServiceWorkerCache() {
+  const isServiceWorkerCacheEnabled = getSCPContext().isServiceWorkerCacheEnabled
+
+  /**
+   * Monkeypatched from
+   * https://github.com/cmap/morpheus.js/blob/8331b8db8696d1bf3255da2261ac729bfc7ea66a/src/io/buffered_reader.js#L36
+   * to enable service worker cache (SWC) in frontend-only SCP development.
+   */
+  window.morpheus.BufferedReader.parse = async function(url, options) {
+    const delim = options.delimiter
+    const regex = new RegExp(delim)
+    const handleTokens = options.handleTokens
+    const complete = options.complete
+    const fetchOptions = {}
+    if (url.headers) {
```

Comment:
Would this benefit from adding a `?` check on url e.g. `if (url?.headers) {`
(not sure if url could be undefined/null)